src/Interpreters/Context.h

```diff
@@ -844,6 +844,9 @@ class Context: public std::enable_shared_from_this<Context>
 
     void shutdown();
 
+    std::vector<std::string> listDisks();
+    void listFiles(const String & name, const String & path, std::vector<String> & file_names);
```

Comment:
You can get disk instance in your calling method, I think this method should be part of the disks tool, not context.